authorize_helper.go

```diff
@@ -22,14 +22,34 @@
 package fosite
 
 import (
+	"html/template"
+	"io"
 	"net/url"
 	"regexp"
+	"strconv"
 	"strings"
+	"time"
+
+	"golang.org/x/net/html"
+	goauth "golang.org/x/oauth2"
 
 	"github.com/asaskevich/govalidator"
 	"github.com/pkg/errors"
 )
 
+var formPostTemplate = template.Must(template.New("form_post").Parse(`<html>
+   <head>
+      <title>Submit This Form</title>
+   </head>
+   <body onload="javascript:document.forms[0].submit()">
+      <form method="post" action="{{ .RedirURL }}">
+         {{ range $key,$value := .Parameters }}
+		<input type="hidden" name="{{$key}}" value="{{index $value 0}}"/>
```

Comment:
Just for completeness (Because other encoding methods like query and fragment supports that), can you loop here over the value (which is a slice) and repeat `<input` with same `key` multiple times? This should never happen. But you never know if somebody directly modifies `Parameters` to add that (given that you defined it as `url.Values`).

Reply:
Also, HTML-escape key and value here.

Reply:
And add test for that.

Reply:
Now the template supports adding multiple values. Also added test case for illegal character encoding.